openedx_external_enrollments/tests/external_enrollments/tests_pathstream_external_enrollments.py

```diff
@@ -29,7 +33,332 @@ def test_str(self):
+    @patch.object(PathstreamExternalEnrollment, '_upload_file')
+    @patch(
+        'openedx_external_enrollments.external_enrollments.pathstream_external_enrollment.ExternalEnrollment',
+        )
+    def test_successful_execute_upload_without_data(
+            self, model_mock, upload_mock, prepare_mock, download_mock, init_s3_mock):
+        """Testing _execute_upload without data, In this case the method
+        must not call any other method.
+
+        Does not have to update any ExternalEnrollment meta
+        """
+        model_mock.objects.filter.return_value = []
+        log = 'There are no new enrollments to update the S3 file.'
+
+        with LogCapture(level=logging.INFO) as log_capture:
+            result = self.base.execute_upload()
+
+            log_capture.check(
+                (self.module, 'INFO', log),
+            )
+
+        init_s3_mock.assert_not_called()
+        download_mock.assert_not_called()
+        prepare_mock.assert_not_called()
+        upload_mock.assert_not_called()
+        model_mock.objects.bulk_update.assert_not_called()
+        self.assertEqual(result[0], True)
+
+    @patch.object(PathstreamExternalEnrollment, '_init_s3')
+    @patch.object(PathstreamExternalEnrollment, '_download_file')
+    @patch.object(PathstreamExternalEnrollment, '_prepare_new_content')
+    @patch.object(PathstreamExternalEnrollment, '_upload_file')
+    @patch(
+        'openedx_external_enrollments.external_enrollments.pathstream_external_enrollment.ExternalEnrollment',
+        )
+    def test_failed_execute_upload(self, model_mock, upload_mock, prepare_mock, download_mock, init_s3_mock):
+        """When any of the methods related to S3 file management fails, it must raise an PathstreamTaskExecutionError
+        which should be capture by _execute_upload in order to log the error."""
+        self.base.client = Mock(spec=boto3.client('s3'))
+        model_mock.objects.filter.return_value = [
+            Mock(
+                controller_name='pathstream',
+                email='test1@gmail.com',
+                meta=[
+                    {
+                        'is_uploaded': False,
+                        'enrollment_data_formated': 'course1,email,2021-07-09 16:53:41.492901,true\n',
+                    },
+                ],
+            ),
+        ]
+        error_operation_msg = ('_download_file', 'error.')
+        log = 'The proccess to update the remote S3 file has failed. Reason: ' + str(error_operation_msg)
+        download_mock.side_effect = PathstreamTaskExecutionError(error_operation_msg[0], error_operation_msg[1])
+
+        with LogCapture(level=logging.ERROR) as log_capture:
+            self.base.execute_upload()
+
+            log_capture.check(
+                (self.module, 'ERROR', log),
+            )
+
+        init_s3_mock.assert_called()
+        prepare_mock.assert_not_called()
+        upload_mock.assert_not_called()
```

Comment:
Missing test case for '_init_s3 has not been called yet or failed.' in execute_upload. @Jacatove